mkosi/kmod.py

```diff
@@ -180,6 +181,15 @@ def gen_required_kernel_modules(
             mods = set(modulesd.rglob("*.ko*"))
         firmware = set()
 
+    # Some firmware dependencies are symbolic links, so the targets for those must be included in the list
+    # of required firmware files too. Intermediate symlinks are not included, and so links pointing to links
+    # results in dangling symlinks in the final image.
+    for fw in firmware.copy():
```

Comment:
Is the call to `copy` even required here?

Reply:
Otherwise this is mutating the set while it is being iterated over, which throws a `RuntimeError: Set changed size during iteration`.